ui-v2/app/services/repository/nspace/enabled.js

```diff
@@ -0,0 +1,51 @@
+import { inject as service } from '@ember/service';
+import { get } from '@ember/object';
+import config from 'consul-ui/config/environment';
+import RepositoryService from 'consul-ui/services/repository';
+
+const modelName = 'nspace';
+export default RepositoryService.extend({
+  router: service('router'),
+  settings: service('settings'),
+  getModelName: function() {
+    return modelName;
+  },
+  findAll: function(configuration = {}) {
+    const query = {};
+    if (typeof configuration.cursor !== 'undefined') {
+      query.index = configuration.cursor;
+    }
+    return this.store.query(this.getModelName(), query);
+  },
+  getUndefinedName: function() {
+    return config.CONSUL_NSPACES_UNDEFINED_NAME;
+  },
+  getActive: function() {
+    let routeParams = {};
+    // this is only populated before the model hook as fired,
+    // it is then deleted after the model hook has finished
+    const infos = get(this, 'router._router.currentState.router.activeTransition.routeInfos');
+    if (typeof infos !== 'undefined') {
+      infos.forEach(function(item) {
+        Object.keys(item.params).forEach(function(prop) {
+          routeParams[prop] = item.params[prop];
+        });
+      });
+    } else {
+      // this is only populated after the model hook has finished
+      //
+      const current = get(this, 'router.currentRoute');
+      if (current) {
+        const nspacedRoute = current.find(function(item, i, arr) {
+          return item.paramNames.includes('nspace');
+        });
+        if (typeof nspacedRoute !== 'undefined') {
+          routeParams.nspace = nspacedRoute.params.nspace;
+        }
+      }
+    }
```

Comment:
I've mentioned this code ^ offline before now. This seems to be the only way to reliably get a part of the `location.hash` from outside of a `Route`. It will also be the place where we eventually write/read from `localStorage` for when we have the `ui/setting` URL which does't include the namespace, similar to how we already do with datacenters.